packages/plugin-typescript/README.md

```diff
@@ -0,0 +1,167 @@
+- _Configuration Errors_: `configuration-errors` - Errors that occur when parsing TypeScript configuration files
+- _Declaration and Language Service Errors_: `declaration-and-language-service-errors` - Errors that occur during TypeScript language service operations
+- _Internal Errors_: `internal-errors` - Errors that occur during TypeScript internal operations
+- _No Implicit Any Errors_: `no-implicit-any-errors` - Errors related to no implicit any compiler option
+- _Unknown Codes_: `unknown-codes` - Errors that do not match any known TypeScript error code
+
+Each audit:
+
+- Checks for specific TypeScript compiler errors and warnings
+- Provides a score based on the number of issues found
+- Includes detailed error messages and locations
+
+Each set is also available as group in the plugin. See more under [Audits and Groups](./docs/audits-and-groups.md).
+
+## Plugin architecture
+
+### Plugin configuration specification
+
+The plugin accepts the following parameters:
+
+| Option       | Type     | Default         | Description                                                                                                                                 |
+| ------------ | -------- | --------------- | ------------------------------------------------------------------------------------------------------------------------------------------- |
+| tsConfigPath | string   | `tsconfig.json` | A string that defines the path to your `tsconfig.json` file                                                                                 |
+| onlyAudits   | string[] | undefined       | An array of audit slugs to specify which documentation types you want to measure. Only the specified audits will be included in the results |
+
+#### TsConfigPath
+
+Optional parameter. The `tsConfigPath` option accepts a string that defines the path to your config file and defaults to `tsconfig.json`.
+
+```js
+typescriptPlugin({
+  tsConfigPath: './tsconfig.json',
+});
+```
+
+#### OnlyAudits
+
+Optional parameter. The `onlyAudits` option allows you to specify which documentation types you want to measure. Only the specified audits will be included in the results. Example:
+
+```js
+typescriptPlugin({
+  onlyAudits: ['no-implicit-any'],
+});
+```
+
+### Optionally set up categories
+
+1. Reference audits (or groups) which you wish to include in custom categories (use `npx code-pushup print-config` to list audits and groups).
+
+Assign weights based on what influence each TypeScript checks should have on the overall category score (assign weight 0 to only include as extra info, without influencing category score).
+
+```ts
+// ...
+categories: [
+  {
+    slug: 'typescript',
+    title: 'TypeScript',
+    refs: [
+      {
+        type: 'audit',
+        plugin: 'typescript',
+        slug: 'no-implicit-any',
+        weight: 2,
+      },
+      {
+        type: 'audit',
+        plugin: 'typescript',
+        slug: 'no-explicit-any',
```

Comment:
This audit doesn't exist in the TypeScript plugin. Please only include valid examples, users may attempt to copy/paste and invalid configurations will just cause confusion.